Bond test is failing on nm-master

The following change of NM is causing this error on the CI test. 

Link: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/commit/190fd9aa9f3fbf5705c2b80b9fc64c89d22b7593

```
tests/integration/bond_test.py::test_replacing_slaves_keeps_mac_of_existing_bond FAILED [  9%]
=================================== FAILURES ===================================
_______________ test_replacing_slaves_keeps_mac_of_existing_bond _______________

bond99_with_eth2 = {'interfaces': [{'link-aggregation': {'mode': 'balance-rr', 'slaves': ['eth1']}, 'name': 'bond99', 'state': 'up', 'type': 'bond'}]}
eth1_up = {'interfaces': [{'ipv4': {'dhcp': False, 'enabled': False}, 'ipv6': {'autoconf': False, 'dhcp': False, 'enabled': False}, 'lldp': {'enabled': False}, 'mac-address': 'CA:45:B4:50:CE:C3', ...}]}

    @pytest.mark.tier1
    def test_replacing_slaves_keeps_mac_of_existing_bond(
        bond99_with_eth2, eth1_up
    ):
        desired_state = bond99_with_eth2
        bond_state = desired_state[Interface.KEY][0]
        eth1_name = eth1_up[Interface.KEY][0][Interface.NAME]
        bond_state[Bond.CONFIG_SUBTREE][Bond.SLAVES] = [eth1_name]
    
        current_state = statelib.show_only((bond_state[Interface.NAME],))
    
        libnmstate.apply(desired_state)
        modified_state = statelib.show_only((bond_state[Interface.NAME],))
>       assert (
            modified_state[Interface.KEY][0][Interface.MAC]
            == current_state[Interface.KEY][0][Interface.MAC]
        )
E       AssertionError: assert 'CA:45:B4:50:CE:C3' == '8A:EE:21:5F:00:8E'
E         - CA:45:B4:50:CE:C3
E         + 8A:EE:21:5F:00:8E

tests/integration/bond_test.py:402: AssertionError
------------------------------ Captured log setup ------------------------------
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Create checkpoint started
2021-03-24 14:16:14 checkpoint.py:133 DEBUG Checkpoint None created for all devices
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Create checkpoint finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Add profile: eth2 started
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Add profile: eth2 finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Activate profile: eth2 started
2021-03-24 14:16:14 connection.py:317 DEBUG Connection activation initiated: dev=eth2, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATING of type NM.ActiveConnectionState>
2021-03-24 14:16:14 connection.py:419 DEBUG Connection activation succeeded: dev=eth2, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATED of type NM.ActiveConnectionState>, dev-state=<enum NM_DEVICE_STATE_ACTIVATED of type NM.DeviceState>, state-flags=<flags NM_ACTIVATION_STATE_FLAG_LAYER2_READY | NM_ACTIVATION_STATE_FLAG_IP4_READY | NM_ACTIVATION_STATE_FLAG_IP6_READY of type NM.ActivationStateFlags>
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Activate profile: eth2 finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/80 started
2021-03-24 14:16:14 checkpoint.py:217 DEBUG Checkpoint /org/freedesktop/NetworkManager/Checkpoint/80 destroyed
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/80 finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Create checkpoint started
2021-03-24 14:16:14 checkpoint.py:133 DEBUG Checkpoint None created for all devices
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Create checkpoint finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Update profile: eth2 started
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Add profile: bond99 started
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Update profile: eth2 finished
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Add profile: bond99 finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Activate profile: bond99 started
2021-03-24 14:16:14 connection.py:317 DEBUG Connection activation initiated: dev=bond99, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATING of type NM.ActiveConnectionState>
2021-03-24 14:16:14 connection.py:419 DEBUG Connection activation succeeded: dev=bond99, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATING of type NM.ActiveConnectionState>, dev-state=<enum NM_DEVICE_STATE_IP_CONFIG of type NM.DeviceState>, state-flags=<flags NM_ACTIVATION_STATE_FLAG_IS_MASTER | NM_ACTIVATION_STATE_FLAG_LAYER2_READY of type NM.ActivationStateFlags>
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Activate profile: bond99 finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Activate profile: eth2 started
2021-03-24 14:16:14 connection.py:317 DEBUG Connection activation initiated: dev=eth2, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATING of type NM.ActiveConnectionState>
2021-03-24 14:16:14 connection.py:419 DEBUG Connection activation succeeded: dev=eth2, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATED of type NM.ActiveConnectionState>, dev-state=<enum NM_DEVICE_STATE_ACTIVATED of type NM.DeviceState>, state-flags=<flags NM_ACTIVATION_STATE_FLAG_IS_SLAVE | NM_ACTIVATION_STATE_FLAG_LAYER2_READY | NM_ACTIVATION_STATE_FLAG_IP4_READY | NM_ACTIVATION_STATE_FLAG_IP6_READY of type NM.ActivationStateFlags>
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Activate profile: eth2 finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/81 started
2021-03-24 14:16:14 checkpoint.py:217 DEBUG Checkpoint /org/freedesktop/NetworkManager/Checkpoint/81 destroyed
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/81 finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Create checkpoint started
2021-03-24 14:16:14 checkpoint.py:133 DEBUG Checkpoint None created for all devices
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Create checkpoint finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Add profile: eth1 started
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Add profile: eth1 finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Activate profile: eth1 started
2021-03-24 14:16:14 connection.py:317 DEBUG Connection activation initiated: dev=eth1, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATING of type NM.ActiveConnectionState>
2021-03-24 14:16:14 connection.py:419 DEBUG Connection activation succeeded: dev=eth1, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATED of type NM.ActiveConnectionState>, dev-state=<enum NM_DEVICE_STATE_ACTIVATED of type NM.DeviceState>, state-flags=<flags NM_ACTIVATION_STATE_FLAG_LAYER2_READY | NM_ACTIVATION_STATE_FLAG_IP4_READY | NM_ACTIVATION_STATE_FLAG_IP6_READY of type NM.ActivationStateFlags>
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Activate profile: eth1 finished
2021-03-24 14:16:14 context.py:147 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/82 started
2021-03-24 14:16:14 checkpoint.py:217 DEBUG Checkpoint /org/freedesktop/NetworkManager/Checkpoint/82 destroyed
2021-03-24 14:16:14 context.py:156 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/82 finished
------------------------------ Captured log call -------------------------------
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Create checkpoint started
2021-03-24 14:16:15 checkpoint.py:133 DEBUG Checkpoint None created for all devices
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Create checkpoint finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Update profile: bond99 started
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Update profile: eth1 started
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Update profile: eth2 started
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Update profile: bond99 finished
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Update profile: eth1 finished
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Update profile: eth2 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Reapply device config: bond99 started
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Reapply device config: bond99 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Reapply device config: eth1 started
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Reapply device config: eth2 started
2021-03-24 14:16:15 device.py:94 DEBUG Device reapply failed on eth1: error=nm-device-error-quark: Can't reapply changes to 'connection.master' setting (3)
Fallback to device activation
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Activate profile: eth1 started
2021-03-24 14:16:15 device.py:94 DEBUG Device reapply failed on eth2: error=nm-device-error-quark: Can't reapply changes to 'connection.master' setting (3)
Fallback to device activation
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Activate profile: eth2 started
2021-03-24 14:16:15 connection.py:317 DEBUG Connection activation initiated: dev=eth1, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATING of type NM.ActiveConnectionState>
2021-03-24 14:16:15 connection.py:317 DEBUG Connection activation initiated: dev=eth2, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATING of type NM.ActiveConnectionState>
2021-03-24 14:16:15 connection.py:419 DEBUG Connection activation succeeded: dev=eth1, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATED of type NM.ActiveConnectionState>, dev-state=<enum NM_DEVICE_STATE_ACTIVATED of type NM.DeviceState>, state-flags=<flags NM_ACTIVATION_STATE_FLAG_IS_SLAVE | NM_ACTIVATION_STATE_FLAG_LAYER2_READY | NM_ACTIVATION_STATE_FLAG_IP4_READY | NM_ACTIVATION_STATE_FLAG_IP6_READY of type NM.ActivationStateFlags>
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Activate profile: eth1 finished
2021-03-24 14:16:15 connection.py:419 DEBUG Connection activation succeeded: dev=eth2, con-state=<enum NM_ACTIVE_CONNECTION_STATE_ACTIVATED of type NM.ActiveConnectionState>, dev-state=<enum NM_DEVICE_STATE_ACTIVATED of type NM.DeviceState>, state-flags=<flags NM_ACTIVATION_STATE_FLAG_LAYER2_READY | NM_ACTIVATION_STATE_FLAG_IP4_READY | NM_ACTIVATION_STATE_FLAG_IP6_READY of type NM.ActivationStateFlags>
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Activate profile: eth2 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/83 started
2021-03-24 14:16:15 checkpoint.py:217 DEBUG Checkpoint /org/freedesktop/NetworkManager/Checkpoint/83 destroyed
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/83 finished
---------------------------- Captured log teardown -----------------------------
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Create checkpoint started
2021-03-24 14:16:15 checkpoint.py:133 DEBUG Checkpoint None created for all devices
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Create checkpoint finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Deactivate profile: eth1 started
2021-03-24 14:16:15 active_connection.py:117 DEBUG Connection deactivation succeeded on eth1
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Deactivate profile: eth1 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Delete profile: id:eth1, uuid:01842f41-f307-4739-937d-a202eb521c62 started
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Delete profile: id:eth1, uuid:01842f41-f307-4739-937d-a202eb521c62 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/84 started
2021-03-24 14:16:15 checkpoint.py:217 DEBUG Checkpoint /org/freedesktop/NetworkManager/Checkpoint/84 destroyed
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/84 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Create checkpoint started
2021-03-24 14:16:15 checkpoint.py:133 DEBUG Checkpoint None created for all devices
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Create checkpoint finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Deactivate profile: bond99 started
2021-03-24 14:16:15 active_connection.py:117 DEBUG Connection deactivation succeeded on bond99
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Deactivate profile: bond99 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Delete profile: id:bond99, uuid:2daa33bf-cf8b-45ea-b483-e18a4eb98143 started
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Delete profile: id:bond99, uuid:2daa33bf-cf8b-45ea-b483-e18a4eb98143 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Delete device: bond99 started
2021-03-24 14:16:15 device.py:144 DEBUG Interface is not real anymore: iface=bond99
2021-03-24 14:16:15 device.py:146 DEBUG Ignored error: g-dbus-error-quark: No such interface 'org.freedesktop.NetworkManager.Device' on object at path /org/freedesktop/NetworkManager/Devices/19 (19)
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Delete device: bond99 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/85 started
2021-03-24 14:16:15 checkpoint.py:217 DEBUG Checkpoint /org/freedesktop/NetworkManager/Checkpoint/85 destroyed
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/85 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Create checkpoint started
2021-03-24 14:16:15 checkpoint.py:133 DEBUG Checkpoint None created for all devices
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Create checkpoint finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Deactivate profile: eth2 started
2021-03-24 14:16:15 active_connection.py:117 DEBUG Connection deactivation succeeded on eth2
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Deactivate profile: eth2 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Delete profile: id:eth2, uuid:e0503f8f-e2e0-4af2-b79e-b11bfa9b2bc7 started
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Delete profile: id:eth2, uuid:e0503f8f-e2e0-4af2-b79e-b11bfa9b2bc7 finished
2021-03-24 14:16:15 context.py:147 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/86 started
2021-03-24 14:16:15 checkpoint.py:217 DEBUG Checkpoint /org/freedesktop/NetworkManager/Checkpoint/86 destroyed
2021-03-24 14:16:15 context.py:156 DEBUG Async action: Destroy checkpoint /org/freedesktop/NetworkManager/Checkpoint/86 finished
```

--- a/tests/integration/bond_test.py
+++ b/tests/integration/bond_test.py
@@ -387,23 +387,6 @@ def test_adding_port_to_empty_bond_doesnt_keep_mac(eth1_up):
         )
 
 
-@pytest.mark.tier1
-def test_replacing_port_keeps_mac_of_existing_bond(bond99_with_eth2, eth1_up):
-    desired_state = bond99_with_eth2
-    bond_state = desired_state[Interface.KEY][0]
-    eth1_name = eth1_up[Interface.KEY][0][Interface.NAME]
-    bond_state[Bond.CONFIG_SUBTREE][Bond.PORT] = [eth1_name]
-
-    current_state = statelib.show_only((bond_state[Interface.NAME],))
-
-    libnmstate.apply(desired_state)
-    modified_state = statelib.show_only((bond_state[Interface.NAME],))
-    assert (
-        modified_state[Interface.KEY][0][Interface.MAC]
-        == current_state[Interface.KEY][0][Interface.MAC]
-    )
-
-
 def test_removing_port_keeps_mac_of_existing_bond(bond99_with_2_port, eth1_up):
     desired_state = bond99_with_2_port
     bond_state = desired_state[Interface.KEY][0]

